src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -264,47 +270,76 @@ export default class QueryEditorUI extends Component<Props, State> {
         return monaco.languages.CompletionItemKind.Field;
       case 'value':
         return monaco.languages.CompletionItemKind.Value;
+      case 'keyword':
+        return monaco.languages.CompletionItemKind.Keyword;
       default:
         return monaco.languages.CompletionItemKind.Text;
     }
   };
 
-  // provideCompletionItems = async (
-  //   model: monaco.editor.ITextModel,
-  //   position: monaco.Position
-  // ): Promise<monaco.languages.CompletionList> => {
-  //   const wordUntil = model.getWordUntilPosition(position);
-  //   const wordRange = new monaco.Range(
-  //     position.lineNumber,
-  //     wordUntil.startColumn,
-  //     position.lineNumber,
-  //     wordUntil.endColumn
-  //   );
-  //   const enhancements = this.props.settings.getQueryEnhancements(this.props.query.language);
-  //   const connectionService = enhancements?.connectionService;
-  //   const suggestions = await this.services.data.autocomplete.getQuerySuggestions({
-  //     query: this.getQueryString(),
-  //     selectionStart: model.getOffsetAt(position),
-  //     selectionEnd: model.getOffsetAt(position),
-  //     language: this.props.query.language,
-  //     indexPatterns: this.state.indexPatterns,
-  //     position,
-  //     connectionService,
-  //   });
-
-  //   return {
-  //     suggestions:
-  //       suggestions && suggestions.length > 0
-  //         ? suggestions.map((s) => ({
-  //             label: s.text,
-  //             kind: this.getCodeEditorSuggestionsType(s.type),
-  //             insertText: s.text,
-  //             range: wordRange,
-  //           }))
-  //         : [],
-  //     incomplete: false,
-  //   };
-  // };
+  private fetchIndexPatterns = async () => {
+    const client = this.services.savedObjects.client;
+    const dataSet = this.queryService.dataSetManager.getDataSet();
+    const title = dataSet?.title;
+
+    const resp = await client.find({
+      type: 'index-pattern',
+      fields: ['title', 'timeFieldName', 'fields'],
+      search: `${title}*`,
+      searchFields: ['title'],
+      perPage: 100,
+    });
+
+    return resp.savedObjects.map((savedObject: any) => ({
+      id: savedObject.id,
+      title: savedObject.attributes?.title,
+      timeFieldName: savedObject.attributes?.timeFieldName,
+      fields: JSON.parse(savedObject.attributes?.fields),
+      type: SIMPLE_DATA_SET_TYPES.INDEX_PATTERN,
+    }));
+  };
+
+  provideCompletionItems = async (
+    model: monaco.editor.ITextModel,
+    position: monaco.Position
+  ): Promise<monaco.languages.CompletionList> => {
+    const enhancements = this.props.settings.getQueryEnhancements(this.props.query.language);
+    const connectionService = enhancements?.connectionService;
```

Comment:
we don't need connectionService

Reply:
this was also left over in early sql changes in `feature/discover-2.0-1`

Reply:
removed